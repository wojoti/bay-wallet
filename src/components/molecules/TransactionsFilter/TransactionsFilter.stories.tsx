import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {TransactionsFilter} from 'components/molecules';

export default {
  title: 'Molecules/TransactionsFilter',
  component: TransactionsFilter,
  argTypes: {
    onFilterPress: {action: 'onFilterPress'},
  },
} as ComponentMeta<typeof TransactionsFilter>;
const Template: ComponentStory<typeof TransactionsFilter> = args => (
  <TransactionsFilter {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  selectedId: 0,
};
