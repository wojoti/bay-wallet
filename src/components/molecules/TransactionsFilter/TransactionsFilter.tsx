import {Button, Container} from 'components/atoms';
import {PropsWithChildren} from 'react';
import style from './TransactionsFilter.style';

export type TransactionsFilterProps = PropsWithChildren<{
  onFilterPress: (id: number) => void;
  selectedId: number;
  testId?: string;
}>;

const TransactionsFilter = ({
  onFilterPress,
  selectedId,
  testId,
}: TransactionsFilterProps) => {
  const onAllPress = () => {
    onFilterPress(0);
  };
  const onExpensePress = () => {
    onFilterPress(1);
  };
  const onIncomePress = () => {
    onFilterPress(2);
  };
  return (
    <Container
      testId={testId}
      direction="row"
      align="center"
      justify="flex-start">
      <Button
        customStyle={style.button}
        customFontStyle={selectedId === 0 ? style.pressed : style.notPressed}
        onPress={onAllPress}
        testId="transactionsfilter-button-all">
        All
      </Button>
      <Button
        customStyle={style.button}
        customFontStyle={selectedId === 1 ? style.pressed : style.notPressed}
        onPress={onExpensePress}
        testId="transactionsfilter-button-expense">
        Expense
      </Button>
      <Button
        customStyle={style.button}
        customFontStyle={selectedId === 2 ? style.pressed : style.notPressed}
        onPress={onIncomePress}
        testId="transactionsfilter-button-income">
        Income
      </Button>
    </Container>
  );
};

export default TransactionsFilter;
