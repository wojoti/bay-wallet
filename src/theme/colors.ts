export const lightPrimary = '#4E439E';
export const primary = '#363062';
export const secondary = 'white';

export const darkGrey = '#595959';
export const grey = '#6B6B6B';
export const vlightGrey = '#D1D1D1';
export const borderGrey = '#E2E2E2';

export const iconGrey = '#BDBDBD';

export const searchGrey = '#F5F5F5';

export const separator = '#F6F6F6';

export const green = '#61C277';
export const lime = '#53E520';
export const red = '#E52020';

export const successGreen = '#3EB290';

export const pink = '#D8B9C3';
export const darkPurple = '#26204E';
export const balanceWhite = '#F8F8FF';

export const marketDarkGrey = '#3D454B';

export const cardYellow = '#FFEF60';

export const buttonGrey = '#EFEFEF';
