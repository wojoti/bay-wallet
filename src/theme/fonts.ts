export const font = 'Poppins-Light';
export const boldFont = 'Poppins-Bold';
