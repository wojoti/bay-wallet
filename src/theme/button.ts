import {StyleSheet} from 'react-native';
import {primary, secondary} from './colors';
const buttonStyle = StyleSheet.create({
  primary: {
    borderWidth: 0,
    backgroundColor: primary,
  },
  secondary: {
    borderColor: primary,
    backgroundColor: secondary,
  },
});

export const buttonTextStyle = StyleSheet.create({
  primary: {
    color: secondary,
  },
  secondary: {
    color: primary,
  },
});

export default buttonStyle;
