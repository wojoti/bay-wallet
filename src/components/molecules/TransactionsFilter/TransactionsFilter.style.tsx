import {StyleSheet} from 'react-native';
import {iconGrey, primary} from 'theme/colors';

const style = StyleSheet.create({
  button: {
    borderWidth: 0,
    marginRight: 20,
  },
  pressed: {
    color: primary,
  },
  notPressed: {
    color: iconGrey,
  },
});
export default style;
