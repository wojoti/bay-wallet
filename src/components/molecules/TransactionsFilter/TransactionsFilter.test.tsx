import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import {iconGrey, primary} from 'theme/colors';
import TransactionsFilter, {
  TransactionsFilterProps,
} from './TransactionsFilter';

const mockOnFilterPress = jest.fn<
  TransactionsFilterProps['onFilterPress'],
  [id: number]
>();
const props: TransactionsFilterProps = {
  testId: 'test-transactionsfilter-id',
  onFilterPress: mockOnFilterPress,
  selectedId: 0,
};

const modifiedProps: TransactionsFilterProps = {
  ...props,
  selectedId: 2,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<TransactionsFilter {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render TransactionsFilter', () => {
  render(<TransactionsFilter {...props} />);
  const TransactionsFilterElement = screen.getByTestId(
    'test-transactionsfilter-id',
  );
  expect(TransactionsFilterElement).toBeOnTheScreen();
});

test('should render TransactionsFilter - All button', () => {
  render(<TransactionsFilter {...props} />);
  const TransactionsFilterElement = screen.getByTestId(
    'test-transactionsfilter-id',
  );
  expect(TransactionsFilterElement).toBeOnTheScreen();
  const buttonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-all',
  );
  expect(buttonElement).toBeOnTheScreen();
});

test('should render TransactionsFilter - All button - onPress', () => {
  render(<TransactionsFilter {...props} />);
  const TransactionsFilterElement = screen.getByTestId(
    'test-transactionsfilter-id',
  );
  expect(TransactionsFilterElement).toBeOnTheScreen();
  const buttonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-all',
  );
  expect(buttonElement).toBeOnTheScreen();
  fireEvent.press(buttonElement);
  expect(mockOnFilterPress).toBeCalledWith(0);
});

test('should render TransactionsFilter - Expense button', () => {
  render(<TransactionsFilter {...props} />);
  const TransactionsFilterElement = screen.getByTestId(
    'test-transactionsfilter-id',
  );
  expect(TransactionsFilterElement).toBeOnTheScreen();
  const buttonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-expense',
  );
  expect(buttonElement).toBeOnTheScreen();
});

test('should render TransactionsFilter - Expense button - onPress', () => {
  render(<TransactionsFilter {...props} />);
  const TransactionsFilterElement = screen.getByTestId(
    'test-transactionsfilter-id',
  );
  expect(TransactionsFilterElement).toBeOnTheScreen();
  const buttonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-expense',
  );
  expect(buttonElement).toBeOnTheScreen();
  fireEvent.press(buttonElement);
  expect(mockOnFilterPress).toBeCalledWith(1);
});

test('should render TransactionsFilter - Income button', () => {
  render(<TransactionsFilter {...props} />);
  const TransactionsFilterElement = screen.getByTestId(
    'test-transactionsfilter-id',
  );
  expect(TransactionsFilterElement).toBeOnTheScreen();
  const buttonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-income',
  );
  expect(buttonElement).toBeOnTheScreen();
});

test('should render TransactionsFilter - Income button - onPress', () => {
  render(<TransactionsFilter {...props} />);
  const TransactionsFilterElement = screen.getByTestId(
    'test-transactionsfilter-id',
  );
  expect(TransactionsFilterElement).toBeOnTheScreen();
  const buttonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-income',
  );
  expect(buttonElement).toBeOnTheScreen();
  fireEvent.press(buttonElement);
  expect(mockOnFilterPress).toBeCalledWith(2);
});

test('should render TransactionsFilter - selected id', () => {
  render(<TransactionsFilter {...props} />);
  const TransactionsFilterElement = screen.getByTestId(
    'test-transactionsfilter-id',
  );
  expect(TransactionsFilterElement).toBeOnTheScreen();
  const allButtonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-all',
  );
  expect(allButtonElement).toBeOnTheScreen();
  const allTextElement = within(allButtonElement).getByTestId('button-text');
  expect(allTextElement).toBeOnTheScreen();
  expect(allTextElement).toHaveStyle({color: primary});

  const expenseButtonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-expense',
  );
  expect(expenseButtonElement).toBeOnTheScreen();
  const expenseTextElement =
    within(expenseButtonElement).getByTestId('button-text');
  expect(expenseTextElement).toBeOnTheScreen();
  expect(expenseTextElement).toHaveStyle({color: iconGrey});

  const incomeButtonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-income',
  );
  expect(incomeButtonElement).toBeOnTheScreen();
  const incomeTextElement =
    within(incomeButtonElement).getByTestId('button-text');
  expect(incomeTextElement).toBeOnTheScreen();
  expect(incomeTextElement).toHaveStyle({color: iconGrey});
});

test('should render TransactionsFilter - different selected id', () => {
  render(<TransactionsFilter {...modifiedProps} />);
  const TransactionsFilterElement = screen.getByTestId(
    'test-transactionsfilter-id',
  );
  expect(TransactionsFilterElement).toBeOnTheScreen();
  const allButtonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-all',
  );
  expect(allButtonElement).toBeOnTheScreen();
  const allTextElement = within(allButtonElement).getByTestId('button-text');
  expect(allTextElement).toBeOnTheScreen();
  expect(allTextElement).toHaveStyle({color: iconGrey});

  const expenseButtonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-expense',
  );
  expect(expenseButtonElement).toBeOnTheScreen();
  const expenseTextElement =
    within(expenseButtonElement).getByTestId('button-text');
  expect(expenseTextElement).toBeOnTheScreen();
  expect(expenseTextElement).toHaveStyle({color: iconGrey});

  const incomeButtonElement = within(TransactionsFilterElement).getByTestId(
    'transactionsfilter-button-income',
  );
  expect(incomeButtonElement).toBeOnTheScreen();
  const incomeTextElement =
    within(incomeButtonElement).getByTestId('button-text');
  expect(incomeTextElement).toBeOnTheScreen();
  expect(incomeTextElement).toHaveStyle({color: primary});
});
