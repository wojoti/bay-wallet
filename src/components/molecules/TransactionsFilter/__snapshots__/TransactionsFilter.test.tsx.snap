// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": undefined,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="test-transactionsfilter-id"
>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 65,
          "justifyContent": "center",
          "opacity": 1,
        },
        false,
        {
          "borderWidth": 0,
          "marginRight": 20,
        },
      ]
    }
    testID="transactionsfilter-button-all"
  >
    <Text
      style={
        [
          {
            "fontFamily": "Poppins-Bold",
            "fontSize": 18,
            "textAlign": "center",
          },
          {
            "color": "#363062",
          },
        ]
      }
      testID="button-text"
    >
      All
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 65,
          "justifyContent": "center",
          "opacity": 1,
        },
        false,
        {
          "borderWidth": 0,
          "marginRight": 20,
        },
      ]
    }
    testID="transactionsfilter-button-expense"
  >
    <Text
      style={
        [
          {
            "fontFamily": "Poppins-Bold",
            "fontSize": 18,
            "textAlign": "center",
          },
          {
            "color": "#BDBDBD",
          },
        ]
      }
      testID="button-text"
    >
      Expense
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 65,
          "justifyContent": "center",
          "opacity": 1,
        },
        false,
        {
          "borderWidth": 0,
          "marginRight": 20,
        },
      ]
    }
    testID="transactionsfilter-button-income"
  >
    <Text
      style={
        [
          {
            "fontFamily": "Poppins-Bold",
            "fontSize": 18,
            "textAlign": "center",
          },
          {
            "color": "#BDBDBD",
          },
        ]
      }
      testID="button-text"
    >
      Income
    </Text>
  </View>
</View>
`;
